semeio/jobs/scripts/fm_pyscal.py

```diff
@@ -0,0 +1,201 @@
+        sys.exit(1)
+
+    if family not in [1, 2]:
+        _logger.error("Family must be either 1 or 2")
+        sys.exit(1)
+
+    if not do_interpolation:
+        int_param_wo = None
+        int_param_go = None
+    if do_interpolation:
+        # Read all key-value pairs from parameters.txt
+        if not os.path.exists(parameters_file_name):
+            _logger.error("%s does not exists", parameters_file_name)
+            sys.exit(1)
+        with open(parameters_file_name) as parameters_file:
+            parameters = parameters_file.readlines()
+        parameter_dict = rm_genkw_prefix(extract_key_value(parameters))
+
+        # Add some magic parameters:
+        parameter_dict.update(MAGIC_CASES)
+        if int_param_wo_name not in parameter_dict:
+            _logger.error(
+                "Requested parameter name %s not found in %s",
+                int_param_wo_name,
+                parameters_file_name,
+            )
+            sys.exit(1)
+        int_param_wo = float(parameter_dict[int_param_wo_name])
+        _logger.info(
+            "Collected %s from parameter.txt with value %f",
+            int_param_wo_name,
+            int_param_wo,
+        )
+        int_param_go = int_param_wo
+
+        if int_param_go_name not in parameter_dict:
+            _logger.error(
+                "Requested parameter name %s not found in %s",
+                int_param_go_name,
+                parameters_file_name,
+            )
+            sys.exit(1)
+        int_param_go = float(parameter_dict[int_param_go_name])
+        if int_param_go_name != int_param_wo_name:
+            _logger.info(
+                "Collected %s from parameter.txt with value %f",
+                int_param_go_name,
+                int_param_go,
+            )
+
+    if sheet_name in ["0", MAGIC_NONE]:
+        # Limitation: If the user names an xls sheet with the string "0"
+        # and it is not the first sheet, it will not be accessible with this
+        # forward model
+        sheet_name = None
+
+    try:
+        pyscalcli.pyscal_main(
+            parametertable=relperm_parameters_file,
+            verbose=True,
+            output=output_filename,
+            sheet_name=sheet_name,
+            int_param_wo=int_param_wo,
+            int_param_go=int_param_go,
+            slgof=slgof == "slgof",
+            family2=family == 2,
+        )
+    except ValueError as e_msg:
+        _logger.error(str(e_msg))
+        sys.exit(1)
+
+
+def rm_genkw_prefix(str_or_dict):
+    """Remove anything before the first colon in a string, or remove
+    the same before every key in a dict"""
+    if isinstance(str_or_dict, six.string_types):
+        if ":" in str_or_dict:
+            parts = str_or_dict.split(":")
+            return ":".join(parts[1:])
+        return str_or_dict
+    if isinstance(str_or_dict, dict):
+        return {rm_genkw_prefix(key): value for key, value in str_or_dict.items()}
+    raise TypeError
```

Comment:
Perhaps a message to follow the exception would be nice?

Reply:
Fixed.